samples/12.customization-minimizable-web-chat/README.md

```diff
@@ -0,0 +1,579 @@
+
+```jsx
+componentDidMount() {
+  !this.props.token && this.props.onFetchToken();
+}
+```
+
+> Note: `onFetchToken()` is a method that will be implemented in `MinimizableWebChat.js`.
+
+
+Let's move on to building the `<MinimizableWebChat>` component.
+
+Import `React`, `createStore`, and `createStyleSet`. Then import your newly made component, `WebChat`.
+
+```jsx
+import React from 'react';
+import { createStore, createStyleSet } from 'botframework-webchat';
+
+import WebChat from './WebChat';
+```
+1. Create the `render()` method
+   1. Create the state object. It should track the following:
+      - If the chat is minimized
+      - If there are new messages
+      - Which side of the window the chat should display on
+      - The store
+      - The bot token, if fetched
+   1. Build the component in the return statement
+      1. Depending on whether it is minimized, the ternary statement will switch between two different renders: minimized and maximized
+      1. Render the `<WebChat>` component when the state is not minimized
+      1. Create a header for the <WebChat> container with two buttons: one to switch sides of the window and the other to minimize the component
+
+```jsx
+render() {
+  const { state: {
+    minimized,
+    newMessage,
+    side,
+    store,
+    styleSet,
+    token
+  } } = this;
+
+  return (
+    <div className="minimizable-web-chat">
+      {
+        minimized ?
+          <button
+            className="maximize"
+            // toggle minimized state
+          >
+            <span className={ token ? 'ms-Icon ms-Icon--MessageFill' : 'ms-Icon ms-Icon--Message' } />
+            {
+              newMessage &&
+                <span className="ms-Icon ms-Icon--CircleShapeSolid red-dot" />
+            }
+          </button>
+        :
+          <div
+            className={ side === 'left' ? 'chat-box left' : 'chat-box right' }
+          >
+            <header>
+              <div className="filler" />
+              <button
+                className="switch"
+                // toggle whether the chat is on the left or right
+              >
+                <span className="ms-Icon ms-Icon--Switch" />
+              </button>
+              <button
+                className="minimize"
+                // toggle minimized state
+              >
+                <span className="ms-Icon ms-Icon--ChromeMinimize" />
+              </button>
+            </header>
+            <WebChat
+              className="react-web-chat"
+              store={ store }
+              styleSet={ styleSet }
+              token={ token }
+            />
+          </div>
+      }
+    </div>
+  );
+}
+```
+
+1. Create the `constructor()`
+   1. In the constructor, create the store with `createStore()`
+   1. Create a filter that sends a message from the bot when the `CONNECT_FULFILLED` action type is detected
```

Comment:
Steps `ii.` & `iii` talk about creating a filter for `CONNECT_FULFILLED` but nowhere in the below jsx can you see that action type anywhere. Seems a little confusing